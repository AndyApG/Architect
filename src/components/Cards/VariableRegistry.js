import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { map, has, get } from 'lodash';
import { TransitionGroup } from 'react-transition-group';
import { Wipe } from '../Transitions';
import { Node, Button } from '../../ui/components';
import { Guided } from '../Guided';
import Guidance from '../Guidance';
import Card from '../Card';
import EdgeIcon from '../EdgeIcon';
import { getProtocol } from '../../selectors/protocol';
import { makeGetUsageForType, makeGetDeleteImpact, makeGetObjectLabel } from '../../selectors/variableRegistry';
import { actionCreators as variableRegistryActions } from '../../ducks/modules/protocol/variableRegistry';
import { actionCreators as dialogsActions } from '../../ducks/modules/dialogs';

const Type = ({ label, link, children, usage, handleDelete }) => (
  <div className="list__item">
    <div className="list__attribute list__attribute--icon">
      <Link to={link}>
        {children}
      </Link>
    </div>
    <div className="list__attribute">
      <h3>
        <Link to={link}>
          {label}
        </Link>
      </h3>
      { usage.length === 0 && <div className="list__tag">unused</div> }
    </div>
    <div className="list__attribute list__attribute--options">
      <Button size="small" color="neon-coral" onClick={handleDelete}>
        Delete
      </Button>
    </div>
  </div>
);

Type.propTypes = {
  label: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  usage: PropTypes.array,
  handleDelete: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

Type.defaultProps = {
  usage: [],
};

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
class VariableRegistry extends Component {
  get buttons() {
    return [
      <Button key="cancel" size="small" color="platinum" onClick={this.handleCancel}>Back</Button>,
    ];
  }

  handleDelete = (entity, type) => {
    const deletedObjects = this.props.getDeleteImpact(entity, type);

    const confirmMessage = (
      <Fragment>
        <p>Are you sure you want to delete {type} {entity}?</p>
        { deletedObjects.length > 0 &&
          <Fragment>
            <p>Because a number of other objects depend on this type, they will also be removed:</p>
            <ul>
              {deletedObjects.map(
                item =>
                  <li>{item.type.toUpperCase()}: {this.props.getObjectLabel(item)}</li>,
              )}
            </ul>
          </Fragment>
        }
      </Fragment>
    );

    this.props.openDialog({
      type: 'Warning',
      title: `Delete ${type} ${entity}`,
      message: confirmMessage,
      onConfirm: () => { this.props.deleteType(entity, type, true); },
      confirmLabel: `Delete ${type} ${entity}`,
    });
  };

  handleCancel = this.props.onComplete;

  renderNode = (node, key) => {
    const nodeColor = get(node, 'color', '');
    const usage = this.props.getUsageForType('node', key);

    return (
      <Wipe key={key}>
        <Type
          link={`${this.props.protocolPath}/registry/node/${key}`}
          label={node.label}
          handleDelete={() => this.handleDelete('node', key)}
          usage={usage}
        >
          <Node label="" color={nodeColor} />
        </Type>
      </Wipe>
    );
  }

  renderEdge = (edge, key) => {
    const edgeColor = `var(--${get(edge, 'color', '')})`;
    const usage = this.props.getUsageForType('edge', key);

    return (
      <Wipe key={key}>
        <Type
          link={`${this.props.protocolPath}/registry/edge/${key}`}
          label={edge.label}
          handleDelete={() => this.handleDelete('edge', key)}
          usage={usage}
        >
          <EdgeIcon color={edgeColor} />
        </Type>
      </Wipe>
    );
  }

  renderEdges() {
    const edges = get(this.props.variableRegistry, 'edge', {});

    if (edges.length === 0) {
      return 'No node types defined';
    }

    return (
      <div className="list">
        <TransitionGroup>
          {map(edges, this.renderEdge)}
        </TransitionGroup>
      </div>
    );
  }

  renderNodes() {
    const nodes = get(this.props.variableRegistry, 'node', {});

    if (nodes.length === 0) {
      return 'No node types defined';
    }

    return (
      <div className="list">
        <TransitionGroup>
          {map(nodes, this.renderNode)}
        </TransitionGroup>
      </div>
    );
  }

  render() {
    const {
      show,
      protocolPath,
    } = this.props;

    return (
      <Card
        show={show}
        buttons={this.buttons}
        onAcknowledgeError={this.handleCancel}
      >
        <div className="editor variable-registry">
          <Guided className="editor__sections">
            <h1>Variable Registry</h1>

            <Guidance contentId="guidance.registry.nodes">
              <div className="editor__section">
                <h2>Node Types</h2>
                <div className="editor__subsection">
                  {this.renderNodes()}
                </div>
                { protocolPath &&
                  <div className="editor__subsection">
                    <Link
                      to={`${protocolPath}/registry/node/`}
                      className="button button--small"
                    >
                      Create new Node type
                    </Link>
                  </div>
                }
              </div>
            </Guidance>

            <Guidance contentId="guidance.registry.edges">
              <div className="editor__section">
                <h2>Edge Types</h2>
                <div className="editor__subsection">
                  {this.renderEdges()}
                </div>
                <div className="editor__subsection">
                  { protocolPath &&
                    <Link
                      to={`${protocolPath}/registry/edge/`}
                      className="button button--small"
                    >
                      Create new Edge type
                    </Link>
                  }
                </div>
              </div>
            </Guidance>
          </Guided>
        </div>
      </Card>
    );
  }
}

VariableRegistry.propTypes = {
  show: PropTypes.bool,
  variableRegistry: PropTypes.shape({
    node: PropTypes.object.isRequired,
    edge: PropTypes.object.isRequired,
  }).isRequired,
  getUsageForType: PropTypes.func.isRequired,
  protocolPath: PropTypes.string,
  onComplete: PropTypes.func,
  deleteType: PropTypes.func.isRequired,
  openDialog: PropTypes.func.isRequired,
  getDeleteImpact: PropTypes.func.isRequired,
  getObjectLabel: PropTypes.func.isRequired,
};

VariableRegistry.defaultProps = {
  protocolPath: null,
  variableRegistry: {
    node: {},
    edge: {},
  },
  show: true,
  onComplete: () => {},
};

const mapStateToProps = (state, props) => {
  const protocol = getProtocol(state);
  const variableRegistry = protocol.variableRegistry;
  const getUsageForType = makeGetUsageForType(state);
  const getDeleteImpact = makeGetDeleteImpact(state);
  const getObjectLabel = makeGetObjectLabel(state);

  return {
    variableRegistry,
    getUsageForType,
    getDeleteImpact,
    getObjectLabel,
    protocolPath: has(props, 'match.params.protocol') ?
      `/edit/${get(props, 'match.params.protocol')}` : null,
  };
};

const mapDispatchToProps = dispatch => ({
  deleteType: bindActionCreators(variableRegistryActions.deleteType, dispatch),
  openDialog: bindActionCreators(dialogsActions.openDialog, dispatch),
});

export { VariableRegistry };

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(VariableRegistry);
