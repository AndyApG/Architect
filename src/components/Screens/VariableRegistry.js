import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { map, get } from 'lodash';
import { TransitionGroup } from 'react-transition-group';
import { Wipe } from '../Transitions';
import { Node, Button, Icon } from '../../ui/components';
import { Guided } from '../Guided';
import Guidance from '../Guidance';
import Card from '../Card';
import Link from '../Link';
import { getProtocol } from '../../selectors/protocol';
import { makeGetUsageForType, makeGetDeleteImpact, makeGetObjectLabel } from '../../selectors/variableRegistry';
import { actionCreators as variableRegistryActions } from '../../ducks/modules/protocol/variableRegistry';
import { actionCreators as dialogsActions } from '../../ducks/modules/dialogs';

const Type = ({ label, category, type, children, usage, handleDelete }) => (
  <div className="simple-list__item">
    <div className="simple-list__attribute simple-list__attribute--icon">
      <Link
        screen="type"
        params={{ category, type }}
      >
        {children}
      </Link>
    </div>
    <div className="simple-list__attribute">
      <h3>
        <Link
          screen="type"
          params={{ category, type }}
        >
          {label}
        </Link>
      </h3>
      { usage.length === 0 && <div className="simple-list__tag">unused</div> }
    </div>
    <div className="simple-list__attribute simple-list__attribute--options">
      <Button size="small" color="neon-coral" onClick={handleDelete}>
        Delete
      </Button>
    </div>
  </div>
);

Type.propTypes = {
  label: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  type: PropTypes.string,
  usage: PropTypes.array,
  handleDelete: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

Type.defaultProps = {
  usage: [],
  type: null,
};

/**
 * This component acts as an index for types. i.e. Nodes and Edges,
 * and links to the EditType.
 */
class VariableRegistry extends Component {
  get buttons() {
    return [
      <Button key="cancel" color="platinum" onClick={this.handleCancel}>Back</Button>,
    ];
  }

  handleDelete = (entity, type) => {
    const deletedObjects = this.props.getDeleteImpact(entity, type);
    const typeName = this.props.variableRegistry[entity][type].name;

    const confirmMessage = (
      <Fragment>
        <p>Are you sure you want to delete {typeName} {entity}?</p>
        { deletedObjects.length > 0 &&
          <Fragment>
            <p>Because a number of other objects depend on this type, they will also be removed:</p>
            <ul>
              {deletedObjects.map(
                item =>
                  <li>{item.type.toUpperCase()}: {this.props.getObjectLabel(item)}</li>,
              )}
            </ul>
          </Fragment>
        }
      </Fragment>
    );

    this.props.openDialog({
      type: 'Warning',
      title: `Delete ${typeName} ${entity}`,
      message: confirmMessage,
      onConfirm: () => { this.props.deleteType(entity, type, true); },
      confirmLabel: `Delete ${typeName} ${entity}`,
    });
  };

  handleCancel = this.props.onComplete;

  renderNode = (node, type) => {
    const nodeColor = get(node, 'color', '');
    const usage = this.props.getUsageForType('node', type);

    return (
      <Wipe key={type}>
        <Type
          category="node"
          type={type}
          label={node.label}
          handleDelete={() => this.handleDelete('node', type)}
          usage={usage}
        >
          <Node label="" color={nodeColor} />
        </Type>
      </Wipe>
    );
  }

  renderEdge = (edge, type) => {
    const edgeColor = get(edge, 'color', '');
    const usage = this.props.getUsageForType('edge', type);

    return (
      <Wipe key={type}>
        <Type
          category="edge"
          type={type}
          label={edge.label}
          handleDelete={() => this.handleDelete('edge', type)}
          usage={usage}
        >
          <Icon name="links" color={edgeColor} />
        </Type>
      </Wipe>
    );
  }

  renderEdges() {
    const edges = get(this.props.variableRegistry, 'edge', {});

    if (edges.length === 0) {
      return 'No node types defined';
    }

    return (
      <div className="simple-list">
        <TransitionGroup>
          {map(edges, this.renderEdge)}
        </TransitionGroup>
      </div>
    );
  }

  renderNodes() {
    const nodes = get(this.props.variableRegistry, 'node', {});

    if (nodes.length === 0) {
      return 'No node types defined';
    }

    return (
      <div className="simple-list">
        <TransitionGroup>
          {map(nodes, this.renderNode)}
        </TransitionGroup>
      </div>
    );
  }

  render() {
    const {
      show,
      state,
    } = this.props;

    return (
      <Card
        show={show}
        state={state}
        buttons={this.buttons}
        onAcknowledgeError={this.handleCancel}
      >
        <Guided>
          <div className="editor variable-registry">
            <div className="editor__window">
              <div className="editor__content">
                <h1 className="editor__heading">Variable Registry</h1>
                <p>
                  Use this screen to create, edit, and manage your node and edge types.
                </p>
                <Guidance contentId="guidance.registry.nodes">
                  <div className="editor__section">
                    <h2>Node Types</h2>
                    <div className="editor__subsection">
                      {this.renderNodes()}
                    </div>
                    <div className="editor__subsection">
                      <Link
                        screen="type"
                        params={{ category: 'node' }}
                      >
                        <Button size="small" icon="add">
                          Create new Node type
                        </Button>
                      </Link>
                    </div>
                  </div>
                </Guidance>

                <Guidance contentId="guidance.registry.edges">
                  <div className="editor__section">
                    <h2>Edge Types</h2>
                    <div className="editor__subsection">
                      {this.renderEdges()}
                    </div>
                    <div className="editor__subsection">
                      <Link
                        screen="type"
                        params={{ category: 'edge' }}
                      >
                        <Button size="small" icon="add">
                          Create new Edge type
                        </Button>
                      </Link>
                    </div>
                  </div>
                </Guidance>
              </div>
            </div>
          </div>
        </Guided>
      </Card>
    );
  }
}

VariableRegistry.propTypes = {
  show: PropTypes.bool,
  state: PropTypes.object,
  variableRegistry: PropTypes.shape({
    node: PropTypes.object.isRequired,
    edge: PropTypes.object.isRequired,
  }).isRequired,
  getUsageForType: PropTypes.func.isRequired,
  onComplete: PropTypes.func,
  deleteType: PropTypes.func.isRequired,
  openDialog: PropTypes.func.isRequired,
  getDeleteImpact: PropTypes.func.isRequired,
  getObjectLabel: PropTypes.func.isRequired,
};

VariableRegistry.defaultProps = {
  variableRegistry: {
    node: {},
    edge: {},
  },
  show: true,
  state: null,
  onComplete: () => {},
};

const mapStateToProps = (state) => {
  const protocol = getProtocol(state);
  const variableRegistry = protocol.variableRegistry;
  const getUsageForType = makeGetUsageForType(state);
  const getDeleteImpact = makeGetDeleteImpact(state);
  const getObjectLabel = makeGetObjectLabel(state);

  return {
    variableRegistry,
    getUsageForType,
    getDeleteImpact,
    getObjectLabel,
  };
};

const mapDispatchToProps = dispatch => ({
  deleteType: bindActionCreators(variableRegistryActions.deleteType, dispatch),
  openDialog: bindActionCreators(dialogsActions.openDialog, dispatch),
});

export { VariableRegistry };

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(VariableRegistry);
