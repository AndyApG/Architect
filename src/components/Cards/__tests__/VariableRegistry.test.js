/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { VariableRegistry } from '../VariableRegistry';

const mockProps = {
  variableRegistry: {
    node: {},
    edge: {},
  },
  getUsageForType: () => {},
  deleteType: () => {},
  protocolPath: '',
  getDeleteImpact: () => {},
  getObjectLabel: () => {},
};

describe('<VariableRegistry />', () => {
  it('can render', () => {
    const component = shallow(<VariableRegistry {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
