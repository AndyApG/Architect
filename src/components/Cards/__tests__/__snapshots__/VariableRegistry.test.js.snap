// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariableRegistry /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VariableRegistry
    deleteType={[Function]}
    onComplete={[Function]}
    protocolPath=""
    show={true}
    variableRegistry={
      Object {
        "edge": Object {},
        "node": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        <Button
          color="platinum"
          content=""
          icon=""
          iconPosition="right"
          onClick={[Function]}
          size="small"
        >
          Back
        </Button>,
      ],
      "children": <div
        className="editor variable-registry"
      >
        <Connect(Guided)
          className="editor__sections"
        >
          <h1>
            Variable Registry
          </h1>
          <Connect(Guided)
            contentId="guidance.registry.nodes"
          >
            <div
              className="editor__section"
            >
              <h2>
                Node Types
              </h2>
              <div
                className="editor__subsection"
              >
                <div
                  className="list"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="div"
                  />
                </div>
              </div>
              
            </div>
          </Connect(Guided)>
          <Connect(Guided)
            contentId="guidance.registry.edges"
          >
            <div
              className="editor__section"
            >
              <h2>
                Edge Types
              </h2>
              <div
                className="editor__subsection"
              >
                <div
                  className="list"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="div"
                  />
                </div>
              </div>
              <div
                className="editor__subsection"
              >
                
              </div>
            </div>
          </Connect(Guided)>
        </Connect(Guided)>
      </div>,
      "enterDelay": 0,
      "enterDuration": 100,
      "exitDelay": 0,
      "exitDuration": 100,
      "onAcknowledgeError": [Function],
      "show": true,
      "style": "wipe",
      "type": "default",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Guided)
          className="editor__sections"
        >
          <h1>
            Variable Registry
          </h1>
          <Connect(Guided)
            contentId="guidance.registry.nodes"
          >
            <div
              className="editor__section"
            >
              <h2>
                Node Types
              </h2>
              <div
                className="editor__subsection"
              >
                <div
                  className="list"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="div"
                  />
                </div>
              </div>
              
            </div>
          </Connect(Guided)>
          <Connect(Guided)
            contentId="guidance.registry.edges"
          >
            <div
              className="editor__section"
            >
              <h2>
                Edge Types
              </h2>
              <div
                className="editor__subsection"
              >
                <div
                  className="list"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="div"
                  />
                </div>
              </div>
              <div
                className="editor__subsection"
              >
                
              </div>
            </div>
          </Connect(Guided)>
        </Connect(Guided)>,
        "className": "editor variable-registry",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h1>
              Variable Registry
            </h1>,
            <Connect(Guided)
              contentId="guidance.registry.nodes"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Node Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                
              </div>
            </Connect(Guided)>,
            <Connect(Guided)
              contentId="guidance.registry.edges"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Edge Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                <div
                  className="editor__subsection"
                >
                  
                </div>
              </div>
            </Connect(Guided)>,
          ],
          "className": "editor__sections",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Variable Registry",
            },
            "ref": null,
            "rendered": "Variable Registry",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="editor__section"
              >
                <h2>
                  Node Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                
              </div>,
              "contentId": "guidance.registry.nodes",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Node Types
                  </h2>,
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>,
                  "",
                ],
                "className": "editor__section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Node Types",
                  },
                  "ref": null,
                  "rendered": "Node Types",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>,
                    "className": "editor__subsection",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />,
                      "className": "list",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "childFactory": [Function],
                        "children": Array [],
                        "component": "div",
                      },
                      "ref": null,
                      "rendered": Array [],
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
                "",
              ],
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="editor__section"
              >
                <h2>
                  Edge Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                <div
                  className="editor__subsection"
                >
                  
                </div>
              </div>,
              "contentId": "guidance.registry.edges",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Edge Types
                  </h2>,
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>,
                  <div
                    className="editor__subsection"
                  >
                    
                  </div>,
                ],
                "className": "editor__section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Edge Types",
                  },
                  "ref": null,
                  "rendered": "Edge Types",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>,
                    "className": "editor__subsection",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />,
                      "className": "list",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "childFactory": [Function],
                        "children": Array [],
                        "component": "div",
                      },
                      "ref": null,
                      "rendered": Array [],
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "editor__subsection",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          <Button
            color="platinum"
            content=""
            icon=""
            iconPosition="right"
            onClick={[Function]}
            size="small"
          >
            Back
          </Button>,
        ],
        "children": <div
          className="editor variable-registry"
        >
          <Connect(Guided)
            className="editor__sections"
          >
            <h1>
              Variable Registry
            </h1>
            <Connect(Guided)
              contentId="guidance.registry.nodes"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Node Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                
              </div>
            </Connect(Guided)>
            <Connect(Guided)
              contentId="guidance.registry.edges"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Edge Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                <div
                  className="editor__subsection"
                >
                  
                </div>
              </div>
            </Connect(Guided)>
          </Connect(Guided)>
        </div>,
        "enterDelay": 0,
        "enterDuration": 100,
        "exitDelay": 0,
        "exitDuration": 100,
        "onAcknowledgeError": [Function],
        "show": true,
        "style": "wipe",
        "type": "default",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Guided)
            className="editor__sections"
          >
            <h1>
              Variable Registry
            </h1>
            <Connect(Guided)
              contentId="guidance.registry.nodes"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Node Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                
              </div>
            </Connect(Guided)>
            <Connect(Guided)
              contentId="guidance.registry.edges"
            >
              <div
                className="editor__section"
              >
                <h2>
                  Edge Types
                </h2>
                <div
                  className="editor__subsection"
                >
                  <div
                    className="list"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component="div"
                    />
                  </div>
                </div>
                <div
                  className="editor__subsection"
                >
                  
                </div>
              </div>
            </Connect(Guided)>
          </Connect(Guided)>,
          "className": "editor variable-registry",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h1>
                Variable Registry
              </h1>,
              <Connect(Guided)
                contentId="guidance.registry.nodes"
              >
                <div
                  className="editor__section"
                >
                  <h2>
                    Node Types
                  </h2>
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>
                  
                </div>
              </Connect(Guided)>,
              <Connect(Guided)
                contentId="guidance.registry.edges"
              >
                <div
                  className="editor__section"
                >
                  <h2>
                    Edge Types
                  </h2>
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>
                  <div
                    className="editor__subsection"
                  >
                    
                  </div>
                </div>
              </Connect(Guided)>,
            ],
            "className": "editor__sections",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Variable Registry",
              },
              "ref": null,
              "rendered": "Variable Registry",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="editor__section"
                >
                  <h2>
                    Node Types
                  </h2>
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>
                  
                </div>,
                "contentId": "guidance.registry.nodes",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2>
                      Node Types
                    </h2>,
                    <div
                      className="editor__subsection"
                    >
                      <div
                        className="list"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />
                      </div>
                    </div>,
                    "",
                  ],
                  "className": "editor__section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Node Types",
                    },
                    "ref": null,
                    "rendered": "Node Types",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="list"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />
                      </div>,
                      "className": "editor__subsection",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />,
                        "className": "list",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "childFactory": [Function],
                          "children": Array [],
                          "component": "div",
                        },
                        "ref": null,
                        "rendered": Array [],
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    "type": "div",
                  },
                  "",
                ],
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="editor__section"
                >
                  <h2>
                    Edge Types
                  </h2>
                  <div
                    className="editor__subsection"
                  >
                    <div
                      className="list"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      />
                    </div>
                  </div>
                  <div
                    className="editor__subsection"
                  >
                    
                  </div>
                </div>,
                "contentId": "guidance.registry.edges",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2>
                      Edge Types
                    </h2>,
                    <div
                      className="editor__subsection"
                    >
                      <div
                        className="list"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />
                      </div>
                    </div>,
                    <div
                      className="editor__subsection"
                    >
                      
                    </div>,
                  ],
                  "className": "editor__section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Edge Types",
                    },
                    "ref": null,
                    "rendered": "Edge Types",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="list"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />
                      </div>,
                      "className": "editor__subsection",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <TransitionGroup
                          childFactory={[Function]}
                          component="div"
                        />,
                        "className": "list",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "childFactory": [Function],
                          "children": Array [],
                          "component": "div",
                        },
                        "ref": null,
                        "rendered": Array [],
                        "type": [Function],
                      },
                      "type": "div",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "editor__subsection",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
